packages/modular-scripts/src/lint.ts

```diff
@@ -0,0 +1,68 @@
+import * as path from 'path';
+import isCI from 'is-ci';
+import { resolveAsBin } from './utils/resolve-as-bin';
+import getModularRoot from './utils/getModularRoot';
+import execSync from './utils/execSync';
+import { getDiffedFiles } from './utils/gitActions';
+import * as logger from './utils/logger';
+import type { LintOptions } from './cli';
+
+export async function lint(
+  options: LintOptions,
+  regexes: string[] = [],
+): Promise<void> {
+  const { all = false, fix = false } = options;
+  const modularRoot = getModularRoot();
+  const lintExtensions = ['.ts', '.tsx', '.js', '.jsx'];
+  let targetedFiles = ['<rootDir>/**/src/**/*.{js,jsx,ts,tsx}'];
+
+  if (!all && !isCI) {
+    const diffedFiles = getDiffedFiles();
+    if (diffedFiles.length === 0) {
+      logger.log(
+        'No diffed files detected. Add the `--all` option to lint the entire codebase',
+      );
+      return;
+    }
+    targetedFiles = diffedFiles
+      .filter((p: string) => lintExtensions.includes(path.extname(p)))
+      .map((p: string) => `<rootDir>/${p}`);
+  }
+
+  const jestEslintConfig = {
+    runner: 'modular-scripts/jest-runner-eslint',
+    displayName: 'lint',
+    rootDir: modularRoot,
+    testMatch: targetedFiles,
+    testPathIgnorePatterns: ['/node_modules/', '/dist/'],
```

Comment:
Good point

Reply:
Looked into it and found that all of the eslintignore patterns are read in by jest-runner-eslint and it will skip the files that patch the globs. Instead of reading in all of the eslintignore files in the repo, we can let jest-runner-eslint skip the files for us.